internal/match/regexp.go

```diff
@@ -0,0 +1,30 @@
+// Copyright 2020 The Go Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package match
+
+// TODO: Move this comment somewhere non-internal later.
+//
+// A license regular expression (LRE) is a pattern syntax intended for
+// describing large English texts such as software licenses, with minor
+// allowed variations. The pattern syntax and the matching are word-based
+// and case-insensitive; punctuation is ignored in the pattern and in
+// the matched text.
+//
+// The valid LRE patterns are:
```

Comment:
This is underspecified. What does `a || b ??` mean, for example? You need to say something about precedence.

Reply:
nvm, this is answered (but elsewhere)